### PR TITLE
Get docs SIte/Md working

### PR DESCRIPTION
**Pull request by Zachiah**

### CODE REVIEW

**Comment by ryansolid** [comment]
I know nothing about mdx and markdown parsing, but its been like a huge debate in the Solid discord and there are multiple different solutions with different tradeoffs. I'm curious if there was a reason for switching it? If I understand this uses vite xdm plugin. I don't know much about these things just curious.

**Comment by nksaraf** [comment]
I think either `xdm` or `@mdx-js/rollup` is fine, they are made by the same person.. @mdx-js just released v2 today.. their v2 was in rc for a while.. Probably better to use the official @mdx-js solution.. I had added xdm when I was experience some issues with certain code-content links. But it didn't work the way I wanted in either.. so neither is better and @mdx-js/mdx is probably gonna be supported properly.

**Comment by Zachiah** [comment]
I fixed the markdown. I'm using '@mdx-js/rollup'

File: docs/package.json [hunk 1]
```diff
@@ -17,14 +17,13 @@
...
+    "solid-mdx": "^0.0.6",
     "solid-start": "workspace:*",
     "solid-start-node": "workspace:*",
     "undici": "^4.12.2",
     "vite": "^2.7.13",
-    "xdm": "^3.3.4"
-  },
-  "dependencies": {
-    "highlight.js": "^11.4.0"
+    "vite-plugin-xdm": "^0.2.0",
```
Comment (nksaraf) [inline docs/package.json:25]: remove this if not being used

File: docs/src/md.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        </ol>
+    ),
+    nav: props => <nav {...props}>{props.children}</nav>,
+    Link,
+    TesterComponent: props => (
+        <p>
+            Remove This Now!!! If you see this it means that markdown custom components does
+            work
+        </p>
+    ),
```
Comment (nksaraf) [inline docs/src/md.tsx:57]: Is this still not working? or can be removed?

File: docs/src/root.tsx [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+import "virtual:windi.css";
+
+import { MDXProvider } from "solid-mdx";
+import { Component } from "solid-js";
+import { Link } from "solid-app-router";
+import Nav from "./Nav";
+import md from "./md";
+
 export default function Root() {
+  console.log({ MDXProvider });
```
Comment (nksaraf) [inline docs/src/root.tsx:14]: remove if unncessary

**Review by nksaraf — COMMENTED** [review]
Looks quite good.. lets see if we can get it deployed somewhere and see what it looks like

File: docs/src/md.tsx (hunk 2 above)
Comment (Zachiah) [reply docs/src/md.tsx:57]: This isn't working. If you look for the usage of `<TesterComponent>`. I just left it so you could see it's not working

File: docs/src/root.tsx (hunk 3 above)
Comment (Zachiah) [reply docs/src/root.tsx]: done

File: docs/package.json (hunk 1 above)
Comment (Zachiah) [reply docs/package.json]: done

**Comment by Zachiah** [comment]
Can't work to much more on this as my endorsement has ended! Hoperfully this helps at least a little

**Comment by ryansolid** [comment]
Thank you so much for all your work on this.

**Pull request merged**